### PR TITLE
api response & error response

### PR DESCRIPTION
**Pull request by mmisztal1980**
Proposal for standard API response in order to enforce testability, handling paging and error handling.

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -209,11 +209,94 @@ If `HASH` is not correct, the server should respond with `401 Unauthorized`.
...
+|Property|Nullable|Type  |Description|
+|--------|--------|------|-----------|
+code     |false   |string|General error category identifier
+message  |false   |string|The error message
+
+*Example error codes:*
+
+|Category      |Description|
+|--------------|-----------|
+err-validation | Data validation error (HTTP 400)
```
Comment (ochococo) [inline README.md]: `err-validation` - `error-validation` ?

File: README.md [hunk 2]
```diff
@@ -209,11 +209,94 @@ If `HASH` is not correct, the server should respond with `401 Unauthorized`.
...
+err-operation  | Invalid operation exception has occured.
+err-server     | Infrastructure error
+wrn-operation  | Warning associated with the operation
```
Comment (ochococo) [inline README.md]: `warning-` - OR why abbreviate!? just use numbers ;) 1 = error, 2 = warning etc ;)

File: README.md [hunk 3]
```diff
@@ -159,7 +159,7 @@ Example:
 			"locality": "New York City",
 			"area": "New York",
 			"country": "US"
-		}
+		},
```
Comment (ochococo) [inline README.md]: 💯

File: README.md (hunk 1 above)
Comment (mmisztal1980) [inline README.md]: I'm open to further discussion on this.

**Pull request merged**